Source/WebKit/UIProcess/Launcher/glib/ProcessLauncherGLib.cpp

```diff
@@ -187,11 +187,15 @@ void ProcessLauncher::launchProcess()
     GRefPtr<GSubprocess> process;
 
 #if OS(LINUX)
-    const char* sandboxEnv = g_getenv("WEBKIT_FORCE_SANDBOX");
     bool sandboxEnabled = m_launchOptions.extraInitializationData.get<HashTranslatorASCIILiteral>("enable-sandbox"_s) == "true"_s;
-
-    if (sandboxEnv)
-        sandboxEnabled = !strcmp(sandboxEnv, "1");
+    const char* sandboxEnv = g_getenv("WEBKIT_FORCE_SANDBOX");
+    if (sandboxEnv) {
+        if (!strcmp(sandboxEnv, "1"))
+            sandboxEnabled = true;
+        else
+            g_warning("WEBKIT_FORCE_SANDBOX no longer allows disabling the sandbox. Use WEBKIT_ALLOW_HACKING_ME=1 instead.");
+    } else if ((sandboxEnv = g_getenv("WEBKIT_ALLOW_HACKING_ME")))
+        sandboxEnabled = strcmp(sandboxEnv, "1");
```

Comment:
Why do we keep the force sandbox env var then? `WEBKIT_ALLOW_HACKING_ME` doesn't sound like a good name either. I would simply use `WEBKIT_DISABLE_SANDBOX` which is useful, not for being hacked, but for debugging purposes.

Reply:
> Why do we keep the force sandbox env var then? 

(a) It can be used to force enable the sandbox. (b) It's useful to provide a warning to people who were previously expecting it to work.

> I would simply use WEBKIT_DISABLE_SANDBOX which is useful, not for being hacked, but for debugging purposes.

This doesn't really convey the badness of disabling the sandbox. I've seen _many_ users running with this environment variable as a workaround for various historical problems without realizing how risky it is to do so.

Reply:
CC @TingPing for opinion

Reply:
> (a) It can be used to force enable the sandbox. (b) It's useful to provide a warning to people who were previously expecting it to work.

why do we need to force the sandbox if it's enabled by default? I guess we only want to keep it for gtk3 builds, env vars are not public api, we don't need to keep compatibility and we are bumping the api version in any case.

> This doesn't really convey the badness of disabling the sandbox. I've seen many users running with this environment variable as a workaround for various historical problems without realizing how risky it is to do so.

Add a g_warning if you want, but the name of the env var should be something like `WEBKIT_DISABLE_SANDBOX` which is exactly what it does.

Reply:
> why do we need to force the sandbox if it's enabled by default? I guess we only want to keep it for gtk3 builds, env vars are not public api, we don't need to keep compatibility and we are bumping the api version in any case.

Oh yeah, that is true.

> Add a g_warning if you want, but the name of the env var should be something like WEBKIT_DISABLE_SANDBOX which is exactly what it does.

Let's just not have the environment variable for now until we can agree on the name. It seems useful for debugging, but if it's not named with care, then it is dangerous.

Reply:
`WEBKIT_ALLOW_HACKING_ME` is a bit over the top...

If you really want to be verbose I agree it should at least be descriptive, like `WEBKIT_DISABLE_SECURITY_FEATURE_SANDBOX` or something. That is pretty loud but also clear.

Reply:
Let's use `WEBKIT_DISABLE_SANDBOX_THIS_IS_DANGEROUS`. A bit longer, but hopefully that's fine.

Reply:
This is missing in the patch in the end, no? I still don't like the name, but I won't block this anymore either.

Reply:
Yes, I will split it out to https://bugs.webkit.org/show_bug.cgi?id=250232 because you asked me to do it in a different bug. Feel free to suggest names there.